File: solutions/erlang/darts/1/src/darts.erl
-module(darts).

-export([score/2]).


score(X, Y) ->
  score(convert_to_polar(X, Y)).

score({Radius, _Theta})
  when (Radius == 0) ->
  10;

score({Radius, _Theta})
  when (Radius =< 5) ->
  5;

score({Radius, _Theta})
  when (Radius =< 10) ->
  1;

score({_Radius, _Theta}) ->
  0.


convert_to_polar(X, Y) ->
  Radius = find_radius(X, Y),
  Theta = unused,
  {Radius, Theta}.

find_radius(X, Y) ->
  math:sqrt((X * X) + (Y * Y)).

File: solutions/csharp/darts/1/Darts.cs
using System;

public static class Darts
{
	private const int _outsideTargetPoints = 0;
	private const int _outerCirclePoints = 1;
	private const int _middlerCirclePoints = 5;
	private const int _innerCirclePoints = 10;

	private const int _outerCircleRadius = 10;
	private const int _middleCircleRadius = 5;
	private const int _innerCircleRadius = 1;

	public static int Score(double x, double y) =>
		Pythagoras.GetDistanceFromCenter(x, y) switch
		{
			<= _innerCircleRadius => _innerCirclePoints,
			<= _middleCircleRadius => _middlerCirclePoints,
			<= _outerCircleRadius => _outerCirclePoints,
			_ => _outsideTargetPoints
		};
}

public static class Pythagoras
{
	public static double GetDistanceFromCenter(double x, double y) =>
		Math.Sqrt(x.Squared() + y.Squared());

	private static double Squared(this double x) =>
		Math.Pow(x, 2);
}